src/ui/public/saved_objects/saved_objects_client.js

```diff
@@ -9,20 +9,91 @@ const join = (...uriComponents) => (
   uriComponents.filter(Boolean).map(encodeURIComponent).join('/')
 );
 
+/**
+ * Interval that requests are batched for
+ * @type {integer}
+ */
+const BATCH_INTERVAL = 100;
+
 export class SavedObjectsClient {
   constructor($http, basePath = chrome.getBasePath(), PromiseCtor = Promise) {
     this._$http = $http;
     this._apiBaseUrl = `${basePath}/api/saved_objects/`;
     this._PromiseCtor = PromiseCtor;
+    this.batchQueue = [];
+  }
+
+  /**
+  * Persists an object
+  *
+  * @param {string} type
+  * @param {object} body - { attributes: {}, id: myId }
+  * @param {object} options
+  * @param {boolean} options.overwrite - defaults to false
+  * @returns {promise}
+  */
+  create(type, body = {}, options = {}) {
+    if (!type || !body.attributes) {
+      return this._PromiseCtor.reject(new Error('requires type and attributes'));
+    }
+
+    const url = this._getUrl([type], options);
+
+    return this._request('POST', url, body);
+  }
+
+  /**
+   * Deletes an object
+   *
+   * @param {string} type
+   * @param {string} id
+   * @returns {promise}
+   */
+  delete(type, id) {
+    if (!type || !id) {
+      return this._PromiseCtor.reject(new Error('requires type and id'));
+    }
+
+    return this._request('DELETE', this._getUrl([type, id]));
+  }
+
+  /**
+   * Search for objects
+   *
+   * @param {object} options
+   * @param {string} options.type
+   * @param {string} options.search
+   * @param {string} options.searchFields - see Elasticsearch Simple Query String
+   *                                        Query field argument for more information
+   * @param {integer} options.page - defaults to 1
+   * @param {integer} options.perPage - defaults to 20
+   * @param {array} option.fields - fields to be returned. Returns all unless defined
+   * @returns {promise}
```

Comment:
Added documentation to each of the methods.

The Saved Object API itself returns `{ id, type, attributes, version }`. When it's passed to the [SavedObject](https://github.com/elastic/kibana/blob/master/src/ui/public/saved_objects/saved_object.js), attributes were previously private and had an underscore. I removed that since it's apparent it's not private. The only remaining thing private to that object is `_version`.

The code you referenced from the Courier SavedObject is slightly confusing since it's translating SavedObject format to the Elasticsearch format. I choose that as opposed to updating every reference to `_source`, `found`,  '_id`, etc.